config/initializers/warn_on_unguarded_column_removals.rb

```diff
@@ -0,0 +1,26 @@
+module ActiveRecord::ConnectionAdapters::SchemaStatements
+  def remove_column_with_column_warning(table_name, column_name, type = nil, options = {})
+    warn_user_about_column_deprecation(table_name, column_name)
+    remove_column_without_column_warning(table_name, column_name, type, options)
+  end
+
+  alias_method_chain :remove_column, :column_warning
+
+  def warn_user_about_column_deprecation(table_name, column_name)
+    begin
+      klass = table_name.to_s.singularize.camelize.constantize
+    rescue NameError
+      puts "WARNING: Couldn't find model for table #{table_name}. I can't tell " +
+           "if column #{column_name} has been deprecated."
+    end
```

Comment:
Do you want a `return` in this rescue block.  Without it, this will give 2 warnings because `klass` will be nil below, and nil doesn't respond to `deprecated_column_list`.